discovery-provider/src/queries/search_queries.py

```diff
@@ -356,20 +350,19 @@ def with_users_added(results):
                     return results
                 return add_users(session, results)
 
-            # Set similarity threshold to be used by % operator in queries.
-            session.execute(sqlalchemy.text(
-                f"select set_limit({minSearchSimilarity});"))
 
             if (searchKind in [SearchKind.all, SearchKind.tracks]):
                 results['tracks'] = with_users_added(track_search_query(
                     session, searchStr, limit, offset, False, is_auto_complete, current_user_id))
-                results['saved_tracks'] = with_users_added(track_search_query(
-                    session, searchStr, limit, offset, True, is_auto_complete, current_user_id))
+                if current_user_id:
```

Comment:
do we need to set all of these fields to [] in the else case?
did you test that this works as expected when a user is not logged in?

Reply:
I tested this on staging with the staging dapp manually pointed at this and it works as intended. Will update the testing description

Reply:
perf 💯